repose-aggregator/components/filters/regex-rbac-filter/src/main/resources/META-INF/schema/config/regex-rbac.xsd

```diff
@@ -0,0 +1,155 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+  Repose
+  _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+  Copyright (C) 2010 - 2015 Rackspace US, Inc.
+  _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+  =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+  -->
+
+
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
+           xmlns:html="http://www.w3.org/1999/xhtml"
+           xmlns:vc="http://www.w3.org/2007/XMLSchema-versioning"
+           xmlns:xerces="http://xerces.apache.org"
+           xmlns:saxon="http://saxon.sf.net/"
+           xmlns="http://docs.openrepose.org/repose/regex-rbac/v1.0"
+           targetNamespace="http://docs.openrepose.org/repose/regex-rbac/v1.0"
+           elementFormDefault="qualified">
+
+    <!-- Elements -->
+    <xs:element name="regex-rbac" type="RegexRbacConfig"/>
+
+    <!-- Types -->
+    <xs:complexType name="RegexRbacConfig">
+        <xs:annotation>
+            <xs:documentation>
+                <html:p>The root config type for the RegEx RBAC filter configuration file.</html:p>
+            </xs:documentation>
+        </xs:annotation>
+
+        <xs:sequence>
+            <xs:element name="delegating" type="DelegatingType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="resources" type="ResourcesType" minOccurs="1" maxOccurs="1"/>
+        </xs:sequence>
+
+        <xs:attribute name="mask-rax-roles-403" type="xs:boolean" use="optional" default="false">
```

Comment:
I know naming this attribute this way makes the transition to the full `api-validator` filter easier, but I just wanted to point out that `rax-roles` is not defined in this context.

`mask-forbidden` or `mask-403` seem like more accurate and intuitive names here.

Reply:
We originally talked about keeping it the same name for consistency of the feature's behavior across all of the filters.

That said, I'm ok changing the name if we want to. This is more of an end user expectation kind of thing.

Reply:
I figured as much, just thought I'd chime in anyway.

In my mind, it's better to name the attribute to make sense in the context it is being used than to name the attribute for consistency across contexts. The reason being that a user is very unlikely to be using both this filter and either the `simple-rbac` or `api-validator` filter, so the consistency isn't buying us anything. Of course, it does buy us something when a user transitions from one filter to another, but I doubt many users would transition from the fuller-featured `simple-rbac` filter or `api-validator` filter to this filter, so it would only be helpful migrating from this filter up to one of those filters. So yes, there is value there, but considering how different this filter is from either of those filters (and thus, how much the user is already going to have to change), I would vote to keep this filter's configuration simpler by not including irrelevant information (i.e., `rax-roles`) in the attribute name. After all, simple is in the name of the filter. ;)

Reply:
Well simple is not in *this* filter's name, that's in the other one.

However, I see the point. I also think consistency in named behavior in the context of *Repose* in general and not just a specific configuration makes sense too. Especially since this is coming to light that this filter will probably be more limited in lifespan and that teams will have to migrate to the full WADL based solutions in the future where the name is used.

As these are just names, it will be easy to do a find/replace on it. I'll wait for the rest of the team to weigh in before making any changes.

Reply:
Gah. That's what I get for adding that line at the last second. My brain got mixed up because we kept tossing the name "super-simple-rbac" around. Anyway, I dropped that sentence to make you look silly.